src/CoreWCF.Primitives/src/CoreWCF/Channels/BinaryMessageEncoderFactory.cs

```diff
@@ -1080,9 +1080,9 @@ public int GetSessionKey(int value)
 
         int GetPrefixOffset(char prefix)
         {
-            if (prefix < 'a' && prefix > 'z')
+            if (prefix < 'a' || prefix > 'z')
```

Comment:
Nice catch. This is a bug which exists in .NET Framework too!! I'll have to take a look and work out what the ramifications are for this bug and get it fixed if it's severe enough.